core/shared/src/main/scala/zio/Queue.scala

```diff
@@ -311,7 +311,7 @@ object Queue {
       }
 
     final def takeUpTo(max: Int): UIO[List[A]] =
-      ZIO.suspend {
+      ZIO.effectSuspend {
```

Comment:
effectSuspendTotal instead?

Reply:
Yes, the `total` variants will _not_ catch exceptions and will return `UIO`.